### ISSUE TITLE
Open url not working for me

### ISSUE THREAD

**Issue by kiki-le-singe** [stevelacy/gulp-open#15]
Hello,

Open an url doesn't work, I don't see what I forgot or not?

This is my code:

```javascript
var gulp = require('gulp');
var open = require('gulp-open');

gulp.task('open', function () {
  var options = {
    url: 'http://localhost:9000',
    app: 'google chrome' // OSX
  };
  
  gulp.src('./app/scripts/index.html')
    .pipe(open('', options));
});
```

**Comment by ywywyul** [comment]
me too

**Comment by stevelacy** [comment]
You need to add google chrome to your OS path:

http://apple.stackexchange.com/questions/95837/find-google-chrome-installation-directory-on-mac-using-terminal

**Comment by anwalkers** [comment]
WIndows works in some cases and in others, for example a web application hosted on IIS, it doesn't... And don't tell me to go to linux or OsX because I can't... (Winging, Sorry)

<pre><code>gulp.task('open', function () {
    return gulp.src('/#')
        .pipe(open('', {
            url: 'http://localhost:80/portal',
            app: 'firefox'
        }));
});</code></pre>

Where http://localhost/portal is a Web Application hosted on IIS and /# is added as a just because it is a Angular JS SPA...

**Comment by anwalkers** [comment]
Trying some tests here....

This works...

<pre><code>gulp.task('open', function(){
	gulp.src('./examples/index.html')
		.pipe(open())
});</code></pre>

This doesn't...

<pre><code>gulp.task('open', function(){
	var config = {
		url: 'http://localhost'
	}
	gulp.src('')
		.pipe(open('', config));
});</code></pre>

Where http://localhost is the default website on IIS...

**Comment by alansouzati** [comment]
Same happened to me, it fixes if I include the `<%file.path%>`, as follows:

```javascript
var gulp = require('gulp');
var open = require('gulp-open');

gulp.task('open', function () {
  var options = {
    url: 'http://localhost:9000'
  };

  gulp.src('./app/scripts/index.html') //this must be a valid and existing path.
    .pipe(open('<%file.path%>', options));
});
```

In my case I was looking to open `http://localhost:9000` as is and as long as `'./app/scripts/index.html'` exists, everything works fine.

Hope it helps...

**Comment by marcellodesales** [comment]
@alansouzati Thanks a lot buddy! That worked!!! Considering all node projects have a project.json, I'm doing the following:

```js
gulp.task('open', function () {
  var options = {
    url: 'http://localhost:9000'
  };

  gulp.src('package.json') //this must be a valid and existing path.
    .pipe(open('<%file.path%>', options));
});
```

thanks!

**Comment by stevelacy** [comment]
@alansouzati, @marcellodesales  want to send a PR to add that to the readme?

Thanks

**Comment by marcellodesales** [comment]
@stevelacy :) Just improved your model and submitted a pull request with an entire fix on the architecture... Supporting any URI, Paths, etc... Simplifying a bit the API as well and allowing users to use either the options or the stream to indicate which file to open.

@alansouzati @anwalkers @ywywyul @kiki-le-singe Could you guys take a look??? It has some validations to verify the cases...

Cheers! 
Marcello

**Comment by alansouzati** [comment]
Good contribution @marcellodesales. I believe the comments from @stevelacy make sense.

**Comment by song940** [comment]
@marcellodesales thanks . that worked.

**Comment by marcellodesales** [comment]
@alansouzati Good... I will work on those changes and send them to the pull request... 

@song940 I'm glad it worked!!! Thanks for verifying it!

**Comment by stevelacy** [comment]
Fixed and published to v1.0.0
github releases shows changelog
